src/c-tips/c-tips.md

```diff
@@ -1,3 +1,219 @@
 # Tips for embedded C developers
 
-> ❌: This section has not yet been written. Please refer to [rust-embedded/book#9](https://github.com/rust-embedded/book/issues/9) for discussion of this section.
+This chapter collects a variety of tips that might be useful to experienced
+embedded C developers looking to start writing Rust. It will especially
+highlight how things you might already be used to in C are different in Rust.
+
+## Preprocessor
+
+In embedded C it is very common to use the preprocessor for a varity of
+purposes, such as:
+
+* Compile-time selection of code blocks with `#ifdef`
+* Compile-time array sizes and computations
+* Macros to simplify common patterns (to avoid function call overhead)
+
+In Rust there is no preprocessor, and so many of these use cases are addressed
+differently. In the rest of this section we cover various alternatives to
+using the preprocessor.
+
+### Compile-Time Code Selection
```

Comment:
I think this section should also (briefly) mention that you can have code conditionally compile based on the target, e.g. `cfg(target_arch)`.

Reply:
Thanks, updated to include `target_arch`.